### PR TITLE
EL-1320: Part 2 - Speed up run_specs

### PR DESCRIPTION
**Pull request by MazOneTwoOne**
[Jira ticket](https://dsdmoj.atlassian.net/browse/EL-1320)

## What changed and why

Our `run_specs` testing Job now runs in parallelised mode, meaning some number of containers (currently 3) share out the tests and run them simultaneously. This allows us to the run tests much more quickly, but not as quickly as (total time if run in one container)/(number of containers), due to setup time.

- Updated image used for `test-executor`, using a GitHub workflow/action based on a new dockerfile. 
- New `run_specs` steps are 
  - download image with chrome and puppeteer installed (as part of the `test-executor`)
  - checkout code
  - [existing steps remain the same]
- Added new CircleCi & GitHub `env_vars` / `secrets`
  - DOCKERHUB_USER_CCQ
  - DOCKERHUB_PAT_CCQ
 - Created a shared DockerHub account

## Checklist

Before you ask people to review this PR:

- Tests and rubocop should be passing
- Branch is generally up to date with main Github - definitely no conflicts
- No unnecessary whitespace changes. These make diffs harder to read and conflicts more likely.
- PR description says *what* changed and *why*, with a link to the JIRA story.
- Diff has been checked for unexpected changes being included.
- Commit messages say why the change was made.

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -8,15 +8,18 @@ executors:
   test-executor:
     resource_class: small
     docker:
-      - image: cimg/ruby:3.2.2-browsers
+      - image: checkclientqualifiesdocker/welcome-to-docker:latest
```
Comment (willc-work) [inline .circleci/config.yml:11]: can we change this `welcome-to-docker` bit and call it something like `circleci_image`?

Comment (MazOneTwoOne) [reply .circleci/config.yml]: @willc-work now updated 👍🏽

File: Dockerfile_browser_tools.dockerfile [hunk 2]
```diff
@@ -0,0 +1,18 @@
+# Build browser tools for test-executor
+# Use the cimg/ruby:3.2.2-browsers image as the base image to extend out
+FROM cimg/ruby:3.2.2-browsers
+
+WORKDIR /app
+
+# Install Chrome
+RUN npx puppeteer browsers install chrome
```
Comment (psweeting1) [inline Dockerfile_browser_tools.dockerfile:8]: You can combine line 8 and line 11 together like so:

```
RUN npx puppeteer browsers install chrome \
    && sudo yarn add puppeteer@21.5.0
```

File: Dockerfile_browser_tools.dockerfile [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+
+# Install Puppeteer with Chromium
+RUN sudo yarn add puppeteer@21.5.0
+
+# Install PDFTK
+RUN sudo add-apt-repository --yes ppa:malteworld/ppa || true
+RUN sudo apt update --allow-unauthenticated || true
+RUN sudo apt install pdftk --allow-unauthenticated || true
```
Comment (psweeting1) [inline Dockerfile_browser_tools.dockerfile:16]: Like wise here we can put them into the same RUN command like so: 

```
RUN sudo add-apt-repository --yes ppa:malteworld/ppa \
    && sudo apt-get update --allow-unauthenticated \
    && sudo apt-get install -y pdftk --allow-unauthenticated
```

Just a question though, don't we want things to fail so we know there is an issue with any of the commands if we use `|| true`?

**Review by psweeting1 — COMMENTED** [review]
Great work! I just added some tidy comments and a question around a shell command

File: Dockerfile_browser_tools.dockerfile (hunk 3 above)
Comment (MazOneTwoOne) [reply Dockerfile_browser_tools.dockerfile:16]: @willc-work - Also agree with getting rid of the `|| true`, I should have removed this now that we have trialed and tested this.

**Pull request merged**